api/prometheus/v1/api.go

```diff
@@ -22,10 +22,12 @@ import (
 	"fmt"
 	"net/http"
 	"strconv"
+	"strings"
 	"time"
 
 	"github.com/prometheus/client_golang/api"
 	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/storage"
```

Comment:
Doesn't this add too much dependency with the whole Prometheus library?
I think the idea is to keep the go client as light as possible.
Can't you just duplicate the struct type?

Reply:
I can duplicate the struct type, but this means even more code duplication. IMO copy/pasting code into this library doesn't really keep it "light", I'd even make the argument that it makes it heavier (since you now have to maintain all of that copy/pasted code). Regardless of my opinion on the matter, if you prefer I can copy/paste that here as well.

Reply:
Yes, we shouldn't depend on Prometheus itself. That'd be a circular dependency at least.

Reply:
@krasi-georgiev do we want to copy/paste the type as is? or do we want this to be `[]string` (since we won't have specific error types to switch on)?